src/enigma2/Epg.cpp

```diff
@@ -40,30 +40,31 @@ bool Epg::Initialise(enigma2::Channels &channels, enigma2::ChannelGroups &channe
   //add an initial EPG data per channel uId, sref and initial EPG
   for (auto& channel : channels.GetChannelsList())
   {
-    EpgChannel newEpgChannel;
+    std::shared_ptr<data::EpgChannel> newEpgChannel = std::make_shared<data::EpgChannel>();
 
-    newEpgChannel.SetRadio(channel->IsRadio());
-    newEpgChannel.SetUniqueId(channel->GetUniqueId());
-    newEpgChannel.SetChannelName(channel->GetChannelName());
-    newEpgChannel.SetServiceReference(channel->GetServiceReference());
+    newEpgChannel->SetRadio(channel->IsRadio());
+    newEpgChannel->SetUniqueId(channel->GetUniqueId());
+    newEpgChannel->SetChannelName(channel->GetChannelName());
+    newEpgChannel->SetServiceReference(channel->GetServiceReference());
 
-    m_epgChannels.emplace_back(new EpgChannel(newEpgChannel));
-    std::shared_ptr<EpgChannel> epgChannel = m_epgChannels.back();
+    m_epgChannels.emplace_back(newEpgChannel);
 
-    m_epgChannelsMap.insert({epgChannel->GetServiceReference(), epgChannel});
-    m_readInitialEpgChannelsMap.insert({epgChannel->GetServiceReference(), epgChannel});
-    m_needsInitialEpgChannelsMap.insert({epgChannel->GetServiceReference(), epgChannel});
+    m_epgChannelsMap.insert({newEpgChannel->GetServiceReference(), newEpgChannel});
+    m_readInitialEpgChannelsMap.insert({newEpgChannel->GetServiceReference(), newEpgChannel});
+    m_needsInitialEpgChannelsMap.insert({newEpgChannel->GetServiceReference(), newEpgChannel});
   }
 
-  int lastScannedIgnoreSuccessCount = round((1 - LAST_SCANNED_INITIAL_EPG_SUCCESS_PERCENT) * m_epgChannels.size());
+  int lastScannedIgnoreSuccessCount = std::round((1 - LAST_SCANNED_INITIAL_EPG_SUCCESS_PERCENT) * m_epgChannels.size());
 
   std::vector<std::shared_ptr<ChannelGroup>> groupList;
-  ChannelGroup newChannelGroup;
-  newChannelGroup.SetRadio(false);
-  newChannelGroup.SetGroupName("Last Scanned"); //Name not important
-  newChannelGroup.SetServiceReference("1:7:1:0:0:0:0:0:0:0:FROM BOUQUET \"userbouquet.LastScanned.tv\" ORDER BY bouquet");
-  newChannelGroup.SetLastScannedGroup(true);
-  groupList.emplace_back(new ChannelGroup(newChannelGroup));
+
+  std::shared_ptr<ChannelGroup> newChannelGroup = std::make_shared<data::ChannelGroup>();
```

Comment:
If you only want to call the default ctor, std::make_shared does not make to much sense. A simple `std::shared_ptr<ChannelGroup> newChannelGroup;` does exactly the same.

Reply:
Obvious now that you say it. Fixed